routemaster/state_machine/api.py

```diff
@@ -101,6 +101,31 @@ def create_label(app: App, label: LabelRef, metadata: Metadata) -> Metadata:
     return metadata
 
 
+def restore_label(app: App, label: LabelRef, metadata: Metadata) -> Metadata:
```

Comment:
this code is untested and might not work -  please just treat it as pseudocode for architecture discussion, for the moment